### PR TITLE
Setup analysis

### PR DESCRIPTION
**Pull request by amelie-renard-sonarsource**

### CODE REVIEW

File: build-analyze.ps1 [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+Add-Type -AssemblyName System.IO.Compression.FileSystem
+[System.IO.Compression.ZipFile]::ExtractToDirectory($path, "$HOME/.sonar")
+$env:Path += ";$HOME/.sonar/sonar-scanner-$SONAR_SCANNER_VERSION-windows\bin"
+
+# Setup the build system
+rm build -Recurse -Force -ErrorAction SilentlyContinue
+mkdir build
+cd build
+cmake ..
+cd ..
```
Comment (arseniy-sonar) [inline build-analyze.ps1:36]: I'm not sure about the syntax in a `ps1` script, but you can do this shorter in a linux shell:
```suggestion
cmake -B build
```

File: build-analyze.ps1 [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+$SONAR_SCANNER_VERSION = "4.6.1.2450" # Find the latest version in the "Windows" link on this page:
+                                      # https://docs.sonarcloud.io/advanced-setup/ci-based-analysis/sonarscanner-cli/
+$BUILD_WRAPPER_OUT_DIR = "build_wrapper_output_directory" # Directory where build-wrapper output will be placed
+
+mkdir $HOME/.sonar
+
+# Download build-wrapper
+$path = "$HOME/.sonar/build-wrapper-win-x86.zip"
+rm build-wrapper-win-x86 -Recurse -Force -ErrorAction SilentlyContinue
+rm $path -Force -ErrorAction SilentlyContinue
```
Comment (arseniy-sonar) [inline build-analyze.ps1:14]: These and l.23, l.24 seem to be there just for the script reentrancy, I think they have no effect for a script that runs in a freshly spawned container. I propose to remove them to simplify the example and not scare users

File: build-analyze.ps1 [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+(New-Object System.Net.WebClient).DownloadFile("$SONAR_SERVER_URL/static/cpp/build-wrapper-win-x86.zip", $path)
+Add-Type -AssemblyName System.IO.Compression.FileSystem
+[System.IO.Compression.ZipFile]::ExtractToDirectory($path, "$HOME/.sonar")
+$env:Path += ";$HOME/.sonar/build-wrapper-win-x86"
+
+# Download sonar-scanner
+$path = "$HOME/.sonar/sonar-scanner-cli-$SONAR_SCANNER_VERSION-windows.zip"
+rm sonar-scanner -Recurse -Force -ErrorAction SilentlyContinue
+rm $path -Force -ErrorAction SilentlyContinue
+[Net.ServicePointManager]::SecurityProtocol = [Net.SecurityProtocolType]::Tls12
```
Comment (arseniy-sonar) [inline build-analyze.ps1:25]: Is this necessary to do again, provided that it is already done on l.15?

File: build-analyze.ps1 [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+(New-Object System.Net.WebClient).DownloadFile("https://binaries.sonarsource.com/Distribution/sonar-scanner-cli/sonar-scanner-cli-$SONAR_SCANNER_VERSION-windows.zip", $path)
+Add-Type -AssemblyName System.IO.Compression.FileSystem
```
Comment (arseniy-sonar) [inline build-analyze.ps1:27]: Same question here: this line repeats l.17, is it needed?

**Review by arseniy-sonar — CHANGES_REQUESTED** [review]
Same as for linux-autotools-otherci-sc PR, I believe it should show how to unshallow the clone.

File: README.adoc [hunk 5]
```diff
@@ -1,3 +1,29 @@
...
+image:{img-qg-status}[Quality Gate Status,link={uri-qg-status}]
+
+*This project is analysed on https://sonarcloud.io/dashboard?id=sonarsource-cfamily-examples_windows-cmake-otherci-sc[SonarCloud]!*
+
+:note-caption: :information_source:
+NOTE: This simple example ignores https://docs.sonarcloud.io/advanced-setup/languages/c-c-objective-c/#analysis-cache[analysis cache] and https://docs.sonarcloud.io/advanced-setup/languages/c-c-objective-c/#parallel-code-scan[multithreaded execution] both of which can boost analysis significantly.
+
+The `build-analyze.ps1` script demonstrates how to run an analysis on a C++ project build. Note that for security reasons, the environment variable `SONAR_TOKEN` is defined in the environement through github secrets. It should not be made public. Github actions are used for demonstration purpose as any CI could be used to run the `build-analyze.ps1` script.
+
+You can take a look at the script link:buid-analyze.ps1[buid-analyze.ps1] of this project to see it in practice.
```
Comment (arseniy-sonar) [inline README.adoc:18]:
```suggestion
You can take a look at the script link:build-analyze.ps1[build-analyze.ps1] of this project to see it in practice.
```

File: build-analyze.ps1 (hunk 3 above)
Comment (amelie-renard-sonarsource) [reply build-analyze.ps1]: Not necessary indeed according to the documentation -> removed

File: build-analyze.ps1 (hunk 4 above)
Comment (amelie-renard-sonarsource) [reply build-analyze.ps1]: Not necessary indeed according to the documentation -> removed

File: build-analyze.ps1 [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+[System.IO.Compression.ZipFile]::ExtractToDirectory($path, "$HOME/.sonar")
+$env:Path += ";$HOME/.sonar/sonar-scanner-$SONAR_SCANNER_VERSION-windows\bin"
```
[the remark this answers is not in the document]
Comment (amelie-renard-sonarsource) [reply build-analyze.ps1]: No point, I suppose it's a typo

File: build-analyze.ps1 [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+                                      # https://docs.sonarcloud.io/advanced-setup/ci-based-analysis/sonarscanner-cli/
+$BUILD_WRAPPER_OUT_DIR = "build_wrapper_output_directory" # Directory where build-wrapper output will be placed
+
+mkdir $HOME/.sonar
+
+# Download build-wrapper
+$path = "$HOME/.sonar/build-wrapper-win-x86.zip"
+[Net.ServicePointManager]::SecurityProtocol = [Net.SecurityProtocolType]::Tls12
+(New-Object System.Net.WebClient).DownloadFile("$SONAR_SERVER_URL/static/cpp/build-wrapper-win-x86.zip", $path)
+Add-Type -AssemblyName System.IO.Compression.FileSystem
```
Comment (arseniy-sonar) [inline build-analyze.ps1:19]:
```suggestion

# Prepare downloading and extraction of build-wrapper and sonar-scanner
[Net.ServicePointManager]::SecurityProtocol = [Net.SecurityProtocolType]::Tls12
Add-Type -AssemblyName System.IO.Compression.FileSystem

# Download build-wrapper
$path = "$HOME/.sonar/build-wrapper-win-x86.zip"
[Net.ServicePointManager]::SecurityProtocol = [Net.SecurityProtocolType]::Tls12
(New-Object System.Net.WebClient).DownloadFile("$SONAR_SERVER_URL/static/cpp/build-wrapper-win-x86.zip", $path)
```

**Review by arseniy-sonar — CHANGES_REQUESTED** [review]

File: build-analyze.ps1 (hunk 7 above)
Comment (amelie-renard-sonarsource) [reply build-analyze.ps1:19]: Ok
(I've modified your suggestion because the line with `Net.ServicePointManager` appeared two times)
```

**Review by arseniy-sonar — APPROVED** [review]

**Pull request merged**